lua/lualine/components/diagnostics.lua

```diff
@@ -0,0 +1,41 @@
+-- Copyright (c) 2020-2021 hoob3rt
+-- MIT license, see LICENSE for more details.
+
+local function diagnostics(options)
+  return function()
+    local error_count, warning_count, info_count = 0,0,0
+    if options.sources~=nil then
+      for _, source in ipairs(options.sources) do
+        if source == 'lsp' then
```

Comment:
I think a nicer way to implement it would be a look up table containing option
Like say:
```lua
local sources = {
  nvim_lsp = function()
    local error_count = vim.lsp.diagnostic.get_count(0, 'Error')
    local warning_count = vim.lsp.diagnostic.get_count(0, 'Warning')
    local info_count = vim.lsp.diagnostic.get_count(0, 'Info')
    return error_count, warning_count, info_count
  end
}
```
Then we can just call the sources with
```lua
local E, W, I = sources[source]()
```

I actualy thought of doing it this way . That way I think code will
be cleaner make it easier to expand . Let me know what you think

Reply:
Yep, we are on the same page here. This is the same way I intend to have it implemented but thanks for pointing that out

Reply:
I'll implement and create a pr soon